base/libgit2/utils.jl

```diff
@@ -34,3 +37,85 @@ function features()
     end
     return res
 end
+
+# Windows authentication prompt
+if is_windows()
+    immutable CREDUI_INFO
+        cbSize::UInt32
+        parent::Ptr{Void}
+        pszMessageText::Ptr{UInt16}
+        pszCaptionText::Ptr{UInt16}
+        banner::Ptr{Void}
+    end
+
+    const CREDUIWIN_GENERIC                 = 0x0001
+    const CREDUIWIN_IN_CRED_ONLY            = 0x0020
+    const CREDUIWIN_ENUMERATE_CURRENT_USER  = 0x0200
+
+    const CRED_PACK_GENERIC_CREDENTIALS     = 0x0004
+
+    const ERROR_SUCCESS                     = 0x0000
+    const ERROR_CANCELLED                   = 0x04c7
+
+    function winprompt(message, caption, default_username; prompt_username = true)
+        # Step 1: Create an encrypted username/password bundle that will be used to set
+        #         the default username (in theory could also provide a default password)
+        credbuf = Array(UInt8, 1024)
+        credbufsize = Ref{UInt32}(sizeof(credbuf))
+        succeeded = ccall((:CredPackAuthenticationBufferW, "Credui.dll"), stdcall, Bool,
```

Comment:
The documentation page I was looking at has this spelling: https://msdn.microsoft.com/en-us/library/windows/desktop/aa374802(v=vs.85).aspx